File: GestionFactura-Backend/GestionFactura.Domain/Entities/Invoice.cs
namespace GestionFactura.Domain.Entities;

public class Invoice
{
    public int Id { get; set; }
    public string InvoiceNumber { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    
    public int ClientId { get; set; }
    public Client Client { get; set; } = null!;

    public int SellerId { get; set; }
    public User Seller { get; set; } = null!;

    public string Status { get; set; } = "Pendiente"; 

    public string PaymentMethod { get; set; } = string.Empty;
    public decimal Subtotal { get; set; }
    public decimal Tax { get; set; }
    public decimal Total { get; set; }

    public ICollection<InvoiceDetail> Details { get; set; } = new List<InvoiceDetail>();
}

File: GestionFactura-Frontend/src/app/core/services/invoice.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

export interface InvoiceDetail {
  productId?: number;
  productCode: string;
  description: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}

export interface InvoiceCreate {
  clientId: number;
  paymentMethod: string;
  subtotal: number;
  tax: number;
  total: number;
  details: InvoiceDetail[];
}

export interface InvoiceSummary {
  id: number;
  invoiceNumber: string;
  date: string;
  clientName: string;
  status: string;
  total: number;
}

@Injectable({
  providedIn: 'root'
})
export class InvoiceService {
  private http = inject(HttpClient);
  private apiUrl = `${environment.apiUrl}/invoices`;

  getInvoices(): Observable<InvoiceSummary[]> {
    return this.http.get<InvoiceSummary[]>(this.apiUrl);
  }

  getInvoice(id: number): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/${id}`);
  }

  createInvoice(invoice: InvoiceCreate): Observable<any> {
    return this.http.post<any>(this.apiUrl, invoice);
  }

  updateInvoice(id: number, invoice: InvoiceCreate): Observable<any> {
    return this.http.put<any>(`${this.apiUrl}/${id}`, invoice);
  }

  deleteInvoice(id: number): Observable<any> {
    return this.http.delete<any>(`${this.apiUrl}/${id}`);
  }
}
